src/main/java/org/dataloader/DataLoader.java

```diff
@@ -470,6 +470,19 @@ private DataLoader(Object batchLoadFunction, DataLoaderOptions options) {
      * @return the promise of the queued load requests
      */
     public CompletableFuture<List<V>> dispatch() {
+        return helper.dispatch().futureList;
+    }
+
+    /**
+     * Dispatches the queued load requests to the batch execution function and returns both the promise of the result
+     * and the number of entries that were dispatched.
+     * <p>
+     * If batching is disabled, or there are no queued requests, then a succeeded promise with no entries dispatched is
+     * returned.
+     *
+     * @return the promise of the queued load requests and the number of entries dispatched.
+     */
+    public DataLoaderHelper.DispatchResult<V> dispatchWithCounts() {
```

Comment:
I am going to accept this PR and fix it up but I will put the changes I will make here for you to see.

Here would would not return a nest class from a helper - I would make it a full class since it now API